### PR TITLE
Allow controlling telemetry via the `ODO_TRACKING_CONSENT` environment variable

### LINKED ISSUE #6253: env variable to enable telementry
/kind user-story

## User Story
As an IDE plugin developer, I want to be able to enable odo telemetry using environment variable siilary how I can disable it using `ODO_DISABLE_TELEMETRY` So that I can enable it, in the same way, I disable it without executing extra odo commands (`odo preference set ConsentTelemetry`)

## Acceptance Criteria 
- [ ] It should be possible to enable telemetry without asking consent questions by setting `ODO_TELEMETRY_TRACKING=true`
- [ ] It should be possible to disable the telemetry setting without asking consent questions `ODO_TELEMETRY_TRACKING=false` (this will be the same behavior as `ODO_DISABLE_TELEMETRY=true`
- [ ] If `ODO_DISABLE_TELEMETRY`  and `ODO_TELEMETRY_TRACKING` are in conflict odo should end with error
- [ ] in our documentation `ODO_DISABLE_TELEMETRY` should be marked as deprecated in favor of `ODO_TELEMETRY_TRACKING`
- [ ] 

/kind user-story
/priority high

### PR DESCRIPTION
**Pull request by rm3l**
**What type of PR is this:**
/kind feature
/area telemetry

**What does this PR do / why we need it:**
See #6253

**Which issue(s) this PR fixes:**
Fixes #6253

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] [Documentation](https://deploy-preview-6258--odo-docusaurus-preview.netlify.app/docs/overview/configure#environment-variables-controlling-odo-behavior)

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by kadel** [comment]
One thing that I have noticed is that if I do `odo preference set ConsentTelemetry false` and then `ODO_TRACKING_CONSENT=yes odo list -v 4` it won't track the `odo list` command. 

We haven't discussed this, but I think that the env variable should override whatever is set in preference.

**Comment by rm3l** [comment]
> One thing that I have noticed is that if I do `odo preference set ConsentTelemetry false` and then `ODO_TRACKING_CONSENT=yes odo list -v 4` it won't track the `odo list` command.
> 
> We haven't discussed this, but I think that the env variable should override whatever is set in preference.
>

Indeed, that makes sense. I'll change that behavior.

**Comment by rm3l** [comment]
> > One thing that I have noticed is that if I do `odo preference set ConsentTelemetry false` and then `ODO_TRACKING_CONSENT=yes odo list -v 4` it won't track the `odo list` command.
> > We haven't discussed this, but I think that the env variable should override whatever is set in preference.
> 
> Indeed, that makes sense. I'll change that behavior.

Done in [cfc64646](https://github.com/redhat-developer/odo/pull/6258/commits/cfc646466b55ff1d5152e7886b562a73832474b9).
I'll rebase the branch later on to fix the conflicts.

**Comment by rm3l** [comment]
```
...
Ran 396 of 396 Specs in 768.518 seconds
SUCCESS! -- 396 Passed | 0 Failed | 0 Pending | 0 Skipped


Ginkgo ran 1 suite in 13m49.56691004s
Test Suite Passed
go run -mod=vendor github.com/onsi/ginkgo/v2/ginkgo  --randomize-all --slow-spec-threshold=120s -timeout 14400s --no-color -nodes=32 tests/e2escenarios
Running Suite: odo e2e scenarios - /go/odo_1/tests/e2escenarios
===============================================================
Random Seed: 1666906779 - will randomize all specs

...
Timed out after 120.000s.
  Expected
      <string>: Hello from Node.js Starter Application!
  to equal
      <string>: Hello from updated Node.js Starter Application!
  In [It] at: /go/odo_1/tests/e2escenarios/e2e_test.go:37
------------------------------


Summarizing 2 Failures:
  [FAIL] E2E Test starting with empty Directory [It] should verify developer workflow from empty Directory
  /go/odo_1/tests/e2escenarios/e2e_test.go:37
  [FAIL] E2E Test starting with non-empty Directory [It] should verify developer workflow from non-empty Directory
  /go/odo_1/tests/e2escenarios/e2e_test.go:37

Ran 6 of 6 Specs in 177.275 seconds
FAIL! -- 4 Passed | 2 Failed | 0 Pending | 0 Skipped
```

E2E Tests fixed by https://github.com/redhat-developer/odo/pull/6260

File: pkg/segment/segment.go [hunk 1]
```diff
@@ -258,14 +268,28 @@ func IsTelemetryEnabled(cfg preference.Client) bool {
...
-		klog.V(4).Infof("Sending telemetry disabled by %s=%t\n", DisableTelemetryEnv, disableTelemetry)
+	trackingConsent := os.Getenv(TrackingConsentEnv)
+
+	if disableTelemetry || trackingConsent == "no" {
+		klog.V(4).Info("Sending telemetry disabled by env variable\n")
 		return false
-	} else if cfg.GetConsentTelemetry() {
+	}
+
+	if trackingConsent == "yes" {
```
Comment (feloy) [inline pkg/segment/segment.go:281]: Could we have functions `trackingConsentEnabled`/`trackingConsentDisabled` instead, so we could rename the values or add aliases later?

Comment (rm3l) [reply pkg/segment/segment.go]: Done - added in [`48669d3` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/48669d394a74b0ee02bb060080aed800cb5d5c4a).

This made me realize that I had few cases not tested, e.g., when the variables are not present in the env. So I added more unit tests in [`75f2321` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/75f23217378f3f83d1b38b11c49611ebbcec41a4)

**Comment by rm3l** [comment]
Had to rebase and force-push to fix the integration tests (#6260).

File: pkg/odo/genericclioptions/runnable.go [hunk 2]
```diff
@@ -80,16 +81,36 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
...
+			klog.V(4).Infof("Skipping telemetry question due to %s=%s\n", segment.TrackingConsentEnv, trackingConsent)
+			klog.V(4).Info("Telemetry is enabled!\n")
+			if err1 := cfg.SetConfiguration(preference.ConsentTelemetrySetting, "true"); err1 != nil {
+				klog.V(4).Info(err1.Error())
+			}
 		} else if disableTelemetry {
+			//lint:ignore SA1019 We deprecated this env var, but until it is removed, we still need to support it
 			klog.V(4).Infof("Skipping telemetry question due to %s=%t\n", segment.DisableTelemetryEnv, disableTelemetry)
 		} else {
 			var consentTelemetry bool
```
Comment (feloy) [inline pkg/odo/genericclioptions/runnable.go:116]: You can also use isTrackingConsentEnabled in this block

Comment (rm3l) [reply pkg/odo/genericclioptions/runnable.go]: Good catch - thanks. Done in [`a4415ec` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/a4415eca1e0856b8486ddb4e4d75934b92876046)

Comment (rm3l) [reply pkg/odo/genericclioptions/runnable.go]: Arf, sorry - I changed in [`8b69040` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/8b69040e10a5d5f743460946a43bbc72f7181e1f). I checked and it should be ok now.

**Comment by feloy** [comment]
/approve
/override ci/prow/v4.11-integration-e2e
All tests pass on IBM Cloud

**Comment by rm3l** [comment]
@kadel Can you please take another look at this PR and approve if it is okay for you? Thanks.

**Pull request merged**